File: 5.py
def sauda(nome):
    print("ola, " + nome + "!")
    sauda2(nome)
    print("preparando pra dizer tchau...")
    tchau()

def sauda2(nome):
    print("como vai, " + nome + "?")

def tchau():
    print ("ok, tchau!")

nome = 'felipe'
sauda('felipe')

File: 5.c
#include <stdio.h>

void sauda(char nome[]);
void sauda2(char nome[]);
void tchau();

void sauda(char nome[]){
    printf("ola, %s!\n", nome);
    sauda2(nome);
    printf("preparando pra dizer tchau...\n");
    tchau();
}

void sauda2(char nome[]){
    printf("como vai, %s?\n", nome);
}

void tchau(){
    printf("ok, tchau!\n");
}


int main(){
    char nome[] = "felipe";
    sauda(nome);

    return 0;
}